learn-arai/app/hooks/useAuth.ts

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+// this react hook responsible for checking if user is logged in.
+import { redirect } from 'next/navigation';
+
+import { useEffect, useState } from 'react';
+
+import { User } from './useUser';
+import { useUser } from './useUser';
+
+export const useAuth = () => {
+    const { addUser, removeUser, user } = useUser();
+    const [isAuthenticated, setIsAuthenticated] = useState<boolean>();
+    const [isSessionExpire, setIsSessionExpire] = useState<boolean>();
+
+    useEffect(() => {
+        checkSession();
+        if (isSessionExpire) {
+            setIsAuthenticated(true);
+        } else {
+            setIsAuthenticated(false);
+        }
+    }, [isSessionExpire]);
+
+    const signIn = (credentials: FormData) => {
+        const response = sendCredentialToServer(credentials);
+
+        return response;
+    };
+
+    const signOut = () => {
+        removeUser();
+    };
+
+    const sendCredentialToServer = async (formData: FormData) => {
+        const response = await fetch('http://localhost:3000/auth/sign-in', {
+            method: 'POST',
```

Comment:
Why having both signIn and sendCredentialToServer? it does the EXACTLY same thing